components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/util/SSOAgentCarbonX509Credential.java

```diff
@@ -0,0 +1,96 @@
+/*
+*  Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.carbon.hostobjects.sso.internal.util;
+
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.core.util.KeyStoreManager;
+
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.cert.X509Certificate;
+
+public class SSOAgentCarbonX509Credential implements SSOAgentX509Credential {
+
+    private PublicKey publicKey = null;
+    private PrivateKey privateKey = null;
+    private X509Certificate entityCertificate = null;
+
+    @Override
+    public PublicKey getPublicKey() {
+        return publicKey;
+    }
+
+    @Override
+    public PrivateKey getPrivateKey() {
+        return privateKey;
+    }
+
+    @Override
+    public X509Certificate getEntityCertificate() {
+        return entityCertificate;
+    }
+
+    public SSOAgentCarbonX509Credential(int tenantId, String tenantDomain)
+            throws Exception {
+
+        readCarbonX509Credentials(tenantId, tenantDomain);
+    }
+
+    protected void readCarbonX509Credentials(int tenantId, String tenantDomain) throws Exception {
```

Comment:
Fixed throwing generic Exception in https://github.com/wso2/carbon-apimgt/pull/1528 (https://github.com/wso2/carbon-apimgt/commit/c6d6cb237bd7654eda952fb2fbd74863e9899cfc)